src/Main.elm

```diff
@@ -1,10 +1,226 @@
+
+
+init : Model
+init =
+    { directory = filter (parentAndTypeAre Nothing Authored) parsedDirectory
+    , parents = []
+    , filterBy = Authored
+    }
+
+
+main : Program () Model Msg
 main =
-    Html.text "There will be app soon!"
+    Browser.sandbox
+        { init = init
+        , view = view
+        , update = update
+        }
+
+
+
+{- Either uses the passed function to filter the directory or returns `typeIs Shared` to avoid
+   tree structure for `Shared` audiences
+-}
+
+
+filterBySelectedFilter : (DirElem -> Bool) -> AudienceType -> (DirElem -> Bool)
+filterBySelectedFilter f at =
+    case at of
+        Shared ->
+            typeIs Shared
+
+        _ ->
+            f
+
+
+update : Msg -> Model -> Model
+update msg model =
+    case msg of
+        OpenFolder folder ->
+            let
+                filteredDir =
+                    filter
+                        (filterBySelectedFilter
+                            (parentAndTypeAre (Just folder.id) model.filterBy)
+                            model.filterBy
+                        )
+                        parsedDirectory
+            in
+            { model
+                | directory = filteredDir
+                , parents = folder.parent :: model.parents
+            }
+
+        GoUp ->
+            let
+                filteredDir =
+                    filter
+                        (filterBySelectedFilter
+                            (parentAndTypeAre (join <| head model.parents) model.filterBy)
+                            model.filterBy
+                        )
+                        parsedDirectory
+            in
+            { model
+                | directory = filteredDir
+                , parents = drop 1 model.parents
+            }
+
+        ChangeFilter at ->
+            let
+                filteredDir =
+                    filter (filterBySelectedFilter (parentAndTypeAre Nothing at) at) parsedDirectory
+            in
+            { directory = filteredDir
+            , parents = []
+            , filterBy = at
+            }
+
+
+showAudienceType : AudienceType -> String
+showAudienceType at =
+    case at of
+        Authored ->
+            "Authored"
+
+        Shared ->
+            "Shared"
+
+        Curated ->
+            "Curated"
+
+
+filterButtonView : AudienceType -> AudienceType -> Html Msg
+filterButtonView at filter =
+    button
+        [ classList [ ( "selected", at == filter ) ]
+        , disabled <| at == filter
+        , onClick <| ChangeFilter at
+        ]
+        [ text <| showAudienceType at ]
+
+
+filterPanelView : Model -> Html Msg
+filterPanelView model =
+    table []
+        [ tr []
+            [ th []
+                [ filterButtonView Authored model.filterBy
+                ]
+            , th []
+                [ filterButtonView Shared model.filterBy
+                ]
+            , th []
+                [ filterButtonView Curated model.filterBy
+                ]
+            ]
```

Comment:
No big deal, but it can be simplified by 
`[Authored, Shared, Curated] |> List map (\t -> 
th []  [ filterButtonView t model.filterBy ] )`

Reply:
Haha, how could I've missed it? 🤦‍♂️